### PR TITLE
[GEOS-10413] Schemaless Features- Allow usage of . separator for PropertyName evaluation - [GEOS-10417] Schemaless PropertyName issue with unbounded nested features

### PR DESCRIPTION
**Pull request by taba90**
[GEOS-10413](https://osgeo-org.atlassian.net/browse/GEOS-10413)

Jira tickets:
- [GEOS-10413](https://osgeo-org.atlassian.net/browse/GEOS-10413)
- [GEOS-10417](https://osgeo-org.atlassian.net/browse/GEOS-10417)
As per the jira tickets this pr:

- fix the simplified property accessor to accept . separated property names.
- fix the simplified property accessor to correctly handle property names pointing to nested features.

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/filter/SchemalessPropertyAccessorFactory.java [hunk 1]
```diff
@@ -101,9 +101,12 @@ private Object walkComplexAttribute(ComplexAttribute complexAttribute, String[]
                 if (result instanceof ComplexAttribute)
                     complexAttribute = (ComplexAttribute) result;
                 else if (result instanceof List) {
+                    boolean lastPart = (i + 1) == path.length;
                     @SuppressWarnings("unchecked")
                     List<Object> attributes = List.class.cast(result);
-                    List<Object> results = walkList(attributes, path, i);
+                    List<Object> results;
+                    if (!lastPart) results = walkList(attributes, path, i);
```
Comment (aaime) [inline src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/filter/SchemalessPropertyAccessorFactory.java:108]: Bit confused about this change... shouldn't it be in walkList? I see walkList/walkComplexAttribute recurse without ever doing this check, which leaves me wondering.

**Review by aaime — COMMENTED** [review]
Seems ok, I have a question.

File: src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/filter/SchemalessPropertyAccessorFactory.java (hunk 1 above)
Comment (taba90) [reply src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/filter/SchemalessPropertyAccessorFactory.java]: you are right, moved it to walk list method.

**Comment by taba90** [comment]
thnaks for you review @aaime feedback applied. I've amended the already existing commit to do a rebase merge and mantain the two commits separated.

**Pull request merged**